addons/ai/XEH_preInit.sqf

```diff
@@ -4,4 +4,16 @@ ADDON = false;
 
 #include "XEH_PREP.hpp"
 
+[QGVAR(updateActiveGroups), {
+    params ["_newGroups"];
+    private _spawnedGroups = GVAR(activeGroups) getOrDefault [_id, []];
+    _spawnedGroups append _newGroups;
+    GVAR(activeGroups) set [_id, _spawnedGroups];
+}] call CBA_fnc_addEventHandler;
+
+[QGVAR(spawnGroups), {
+    params ["_groups"];
+    [_groups] call FUNC(spawnGroups);
+}] call CBA_fnc_addEventHandler;
```

Comment:
Wrap in an `if (isServer || {!hasInterface})` check

Also inside the eventhandler, no need for params, just do `call FUNC(spawnGroups);` as `_this` is passed implicitly